r/R/table.R

```diff
@@ -202,7 +210,27 @@ Table$create <- function(..., schema = NULL) {
 
 #' @export
 as.data.frame.Table <- function(x, row.names = NULL, optional = FALSE, ...) {
-  Table__to_dataframe(x, use_threads = option_use_threads())
+  df <- Table__to_dataframe(x, use_threads = option_use_threads())
+
+  if (!is.null(r_metadata <- x$metadata$r)) {
+    r_metadata <- .arrow_unserialize(r_metadata)
+
+    df_metadata <- r_metadata[[1L]]
```

Comment:
IMO using a named list would be clearer than relying on position in an unnamed list

Reply:
Thanks